File: app/src/main/java/com/islam/basepropject/ui/MainActivity.java
package com.islam.basepropject.ui;

import com.islam.basepropject.R;
import com.islam.basepropject.project_base.base.POJO.NavigationType;
import com.islam.basepropject.project_base.base.activities.BaseNavigationActivity;
import com.islam.basepropject.project_base.common.ui.settings.SettingsFragment;
import com.islam.basepropject.ui.ExFetchData.Fragment1;
import com.islam.basepropject.ui.ExRecyclerView.Fragment3;
import com.islam.basepropject.ui.ExPaging.ExFragment;
import com.islam.basepropject.ui.ExViewPager.Fragment4;

public class MainActivity extends BaseNavigationActivity {


    @Override
    public void onLaunch() {
        initContentView(R.layout.activity_main);
        initNavigation(new Class[]{Fragment1.class, Fragment3.class, Fragment4.class, ExFragment.class, SettingsFragment.class},
                new int[]{R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow,R.id.nav_tools,R.id.nav_settings},
                NavigationType.DrawerNavigation);
    }


}

File: app/src/main/java/com/islam/basepropject/ui/MainActivity.kt
package com.islam.basepropject.ui

import com.islam.basepropject.R
import com.islam.basepropject.project_base.POJO.NavigationType
import com.islam.basepropject.project_base.base.activities.BaseNavigationActivity
import com.islam.basepropject.project_base.common.ui.settings.SettingsFragment
import com.islam.basepropject.ui.examples.ExFetchData.Fragment1
import com.islam.basepropject.ui.examples.ExPaging.ExFragment
import com.islam.basepropject.ui.examples.ExRecyclerView.Fragment3
import com.islam.basepropject.ui.examples.Fragment4

class MainActivity : BaseNavigationActivity() {

    override val layoutId = R.layout.activity_main

    override val navigationType = NavigationType.DrawerNavigation

    override val menuIdsListWithFragment: MutableList<Pair<Int, Class<*>>>
        get() = mutableListOf<Pair<Int, Class<*>>>().run {
                add(Pair(R.id.nav_home, Fragment1::class.java))
                add(Pair(R.id.nav_gallery, Fragment3::class.java))
                add(Pair(R.id.nav_slideshow, Fragment4::class.java))
                add(Pair(R.id.nav_tools, ExFragment::class.java))
                add(Pair(R.id.nav_settings, SettingsFragment::class.java))
                this
            }


}

